### DISCUSSION TITLE
Finding it difficult to install nada-dsl

### DISCUSSION THREAD

**Discussion by ScapeXYZ** [NillionNetwork/discussions#133] [Bugs]
I'm unable to install nada-dsl
I'm getting this error
`pip install --upgrade nada-dsl
ERROR: Could not find a version that satisfies the requirement nada-dsl (from versions: none)
ERROR: No matching distribution found for nada-dsl`
I need a help

**Comment by navasvarela** [comment]
I have just tried this command locally and it works for me. Could you share more details about your environment and if you're trying to run this command in a python project?

What is your Python version?

**Reply by ScapeXYZ** [reply]
Python 3.13

**Reply by ScapeXYZ** [reply]
What more details can I share

**Reply by navasvarela** [reply]
Can you share the output of `pip --version`?

**Reply by ScapeXYZ** [reply]
pip --version
Traceback (most recent call last):
  File "/home/temitope/nada/.venv/bin/pip", line 5, in <module>
    from pip._internal.cli.main import main
  File "/home/temitope/nada/.venv/lib/python3.8/site-packages/pip/_internal/cli/main.py", line 11, in <module>
    from pip._internal.cli.autocompletion import autocomplete
  File "/home/temitope/nada/.venv/lib/python3.8/site-packages/pip/_internal/cli/autocompletion.py", line 10, in <module>
    from pip._internal.cli.main_parser import create_main_parser
  File "/home/temitope/nada/.venv/lib/python3.8/site-packages/pip/_internal/cli/main_parser.py", line 9, in <module>
    from pip._internal.build_env import get_runnable_pip
  File "/home/temitope/nada/.venv/lib/python3.8/site-packages/pip/_internal/build_env.py", line 18, in <module>
    from pip._internal.cli.spinners import open_spinner
  File "/home/temitope/nada/.venv/lib/python3.8/site-packages/pip/_internal/cli/spinners.py", line 9, in <module>
    from pip._internal.utils.logging import get_indentation
  File "/home/temitope/nada/.venv/lib/python3.8/site-packages/pip/_internal/utils/logging.py", line 29, in <module>
    from pip._internal.utils.misc import ensure_dir
  File "/home/temitope/nada/.venv/lib/python3.8/site-packages/pip/_internal/utils/misc.py", line 41, in <module>
    from pip._internal.locations import get_major_minor_version
  File "/home/temitope/nada/.venv/lib/python3.8/site-packages/pip/_internal/locations/__init__.py", line 66, in <module>
    from . import _distutils
  File "/home/temitope/nada/.venv/lib/python3.8/site-packages/pip/_internal/locations/_distutils.py", line 20, in <module>
    from distutils.cmd import Command as DistutilsCommand
ModuleNotFoundError: No module named 'distutils.cmd'

**Reply by navasvarela** [reply]
It seems like your Python environment is pretty broken. You need to fix it. Also it is using Python 3.8, which is the root cause of the problem. You cannot install nada-dsl on Python < 3.10. My recommendation is that you fix your local Python environment. I'd advise you to look into using pyenv as a way of managing different Python versions.

**Reply by ScapeXYZ** [reply]
Sorry.
This is where I had being getting error 

Run nillion-devnet in another terminal window
Reload your Next.js app and you should be able to login, store a value and retrieve a value.
Can you guide me from there

**Reply by crypblizz8** [reply]
Is your python up to date + nada all setup now though? That is the first step.

**Comment by ScapeXYZ** [comment]
I had passed through that error 
Presently on NADA build 
`[[programs]]`
`path = "src/secret_addition.py"`
`name = "secret_addition"`
`prime_size = 128`
nada-project.toml file 

Error I'm getting 
nada build
Error:
   0: TOML parse error at line 1, column 1
   0:   |
   0: 1 | [[programs]]
   0:   | ^
   0: missing field `name`
   0:

Location:
   tools/nada/src/nada_project_toml.rs:67

Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.

**Reply by ScapeXYZ** [reply]
@crypblizz8

**Reply by ScapeXYZ** [reply]
Please need your help on NADA build

**Reply by navasvarela** [reply]
Could you share the full listing of `nada-project.toml`?  It is important that `nada-project.toml` is in the correct format. We have a command `nada init PROJECT_NAME` that initialises a nada project with a default program. This is the recommended way to bootstrap a new project since it generates the project structure and initial configuration for you.

**Comment by ScapeXYZ** [comment]
![IMG_20241022_151400](https://github.com/user-attachments/assets/4954a403-10cd-4801-aee9-6ba28d06ce2a)

**Comment by ScapeXYZ** [comment]
Sorry 
I had start over with GitHub codespace now 
Running Nillion-devnet

`nillion-devnet
ℹ️ cluster id is 9e68173f-9c23-4acc-ba81-4f079b639964
ℹ️ using 256 bit prime
ℹ️ storing state in /tmp/.tmp02r7ov (23.16Gbs available)
🏃 starting nilchain node in: /tmp/.tmp02r7ov/nillion-chain
Error: failed to init chain

Caused by:
    exited with status: exit status: 1`

This is the error I'm facing now @navasvarela

**Reply by crypblizz8** [reply]
What did you do to solve this @ScapeXYZ
